utils/io/content/contentreader.go

```diff
@@ -0,0 +1,158 @@
+package content
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+type ContentReader struct {
+	// Response data file path.
+	filePath, arrayKey string
+	// The objects from the response data file are being pushed to the data channel.
+	dataChannel chan map[string]interface{}
+	buffer      []map[string]interface{}
+	errorsQueue *utils.ErrorsQueue
+	once        *sync.Once
+}
+
+func NewContentReader(filePath string, arrayKey string) *ContentReader {
+	self := ContentReader{}
+	self.filePath = filePath
+	self.arrayKey = arrayKey
+	self.dataChannel = make(chan map[string]interface{}, 50000)
+	self.errorsQueue = utils.NewErrorsQueue(50000)
+	self.once = new(sync.Once)
+	return &self
+}
+
+func (rr *ContentReader) ArrayKey(arrayKey string) *ContentReader {
+	rr.arrayKey = arrayKey
+	return rr
+}
+
+func (rr *ContentReader) NextRecord(recordOutput interface{}) error {
+	rr.once.Do(func() {
+		go func() {
+			rr.run()
+		}()
+	})
+	record, ok := <-rr.dataChannel
+	if !ok {
+		return errorutils.CheckError(io.EOF)
+	}
+	data, _ := json.Marshal(record)
+	return errorutils.CheckError(json.Unmarshal(data, recordOutput))
+}
+
+func (rr *ContentReader) Reset() {
+	rr.dataChannel = make(chan map[string]interface{}, 50000)
+	rr.once = new(sync.Once)
+}
+
+func (rr *ContentReader) Close() error {
+	if rr.filePath != "" {
+		return errorutils.CheckError(os.Remove(rr.filePath))
+	}
```

Comment:
I think we should set rr.filePath to en empty string here, because fo the code in the SetFilePath function.